test/fixtures/minio-fixture/src/main/java/org/elasticsearch/test/fixtures/minio/MinioFixtureTestContainer.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.test.fixtures.minio;
+
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.images.builder.ImageFromDockerfile;
+
+public class MinioFixtureTestContainer implements TestRule {
+
+    private static final int servicePort = 9000;
+    private GenericContainer<?> container;
+
+    private GenericContainer<?> createContainer() {
+        return new GenericContainer<>(
```

Comment:
Would it be better to simply extend `GenericContainer` and then do this stuff in the constructor? That would avoid having to explicitly implement `TestRule` and `apply()`.